predicators/approaches/grammar_search_invention_approach.py

```diff
@@ -1025,6 +1028,11 @@ def learn_from_offline_dataset(self, dataset: Dataset) -> None:
             score_function = create_score_function(
                 CFG.grammar_search_score_function, self._initial_predicates,
                 atom_dataset, candidates, self._train_tasks)
+            self._learned_predicates = set(
+                p for p in candidates.keys() if p.name in [
+                    "((0:robot).fingers<=[idx 0]0.5)", "Cooked0", "Whole0",
+                    "Diced0", "Cut0", "Sliced0", "Shredded0", "Chopped0"
+                ])
```

Comment:
I'm confused: the point of adding this here is to avoid hill-climbing right? but from the code, it looks like hill-climbing is still being run? 
Also, is there a reason we couldn't just make a debug grammar here instead of doing this?

Reply:
My first reply: I was gonna remove this before the final PR -- I just have this here while I test the PR. 

But also -- remember that the implementation of debug grammar still runs hill climbing -- just on that specific hardcoded pool of predicates -- so it's still slower than just choosing the predicates you want right away. So, I didn't use the debug grammar for speed reasons. 

The right solution is to have two different ways the debug grammar is used -- (1) one where it actually runs hill climbing on the debug grammar pool, another (2) where it just takes the debug grammar predicates directly. 

We'd use (1) when we want to actually test the hill climbing search, and (2) when hill climbing itself is not important.

Reply:
oh good point. maybe worth quickly implementing this "right" solution? I'm sure it'll be useful in future envs as well!

Reply:
Will do.